accessor.go

```diff
@@ -55,14 +55,33 @@ func (sa *ShardAccessor) Shard() shard.Key {
 }
 
 func (sa *ShardAccessor) Read(p []byte) (int, error) {
+	sa.lk.Lock()
+	defer sa.lk.Unlock()
+
+	if sa.mmapr == nil {
+		if f, ok := sa.data.(*os.File); ok {
+			if mmapr, err := mmap.Open(f.Name()); err != nil {
+				log.Warnf("failed to mmap reader of type %T: %s; using reader as-is", sa.data, err)
+			} else {
+				sa.mmapr = mmapr
+			}
+		}
+	}
+
+	if sa.mmapr != nil {
+		return sa.mmapr.ReadAt(p, 0)
+	}
```

Comment:
* `mmap.ReadAt` has no internal state on how much data was read out of it. 
	* Basically, calling `Read` several times on the Accessor will always give you the same result instead of progressing until io.EOF as with usual Readers.
	* So to make current solution work, there should be some io.Reader implementation around io.ReaderAt, which should be allocated per each concurrent reader.

* `mmap` is usually great when you need random access over the file, like in the Blockstore case below. In the sequential read case, I wonder if there are any benefits compared to a simple read to the userspace and keeping it cached there for future accesses. 
	* Still, I see the motivation here - reusal of mmaped file in case it is also accessed from Blockstore.
	* It's likely, though, that keeping two copies: one for Blockstore and one for regular reads will be faster compared to one shared mmaped file and easier to implement, but still requires further thoughts.

Reply:
Since we're already using mmap for the Blockstore call to serve LNs, the pages are mapped, and having a separate copy would lead to unnecessary overhead when switching between the allocations. Though multiple reads would always start from zero, the reuse of the existing mmaped file is more efficient and already allows us to access the data like an array. Your first point about calling Read several times though is definitely a concern, I would suggest the simplest solution is returning a `ReaderAt` and having the caller keep track of how many bytes it is reading.

Reply:
In the private discussion, we agreed to keep shared mmap and make a `Reader()` method on Accessor that enables getting readers